src/ingest-pipeline/misc/tools/extract_for_review.py

```diff
@@ -43,7 +43,17 @@ def main():
                     dataset_name = ''
                     if _dataset.prop_dct.get('lab_dataset_id'):
                         dataset_name = _dataset.prop_dct.get('lab_dataset_id')
-                    if dataset.prop_dct.get('provider_info'):
+                    elif _dataset.prop_dct.get('provider_info'):
+                        if dataset_name == '':
+                            dataset_name = _dataset.prop_dct.get('provider_info')
+                        else:
+                            dataset_name += '/' + _dataset.prop_dct.get('provider_info')
+                    if dataset.prop_dct.get('lab_dataset_id'):
+                        if dataset_name == '':
+                            dataset_name = dataset.prop_dct.get('lab_dataset_id')
+                        else:
+                            dataset_name += '/' + dataset.prop_dct.get('lab_dataset_id')
+                    elif dataset.prop_dct.get('provider_info'):
```

Comment:
This whole block of code is very confusion. I had to graph out all of the conditionals, plus using `dataset_name` as a conditional as well as value storage made it worse, I'm afraid.

So, after graphing it out, I came up with this skeleton. I believe this code should achieve the same result, but may be more readable. What do you think?

```
if dataset.parents is not None:
    for parent_uuid, parent_dataset in dataset.parents.items():
        parent_component = (
                parent_dataset.prop_dct.get('lab_dataset_id')
                or parent_dataset.prop_dct.get('provider_info')
        )
        child_component = (
                dataset.prop_dct.get('lab_dataset_id')
                or dataset.prop_dct.get('provider_info')
        )
        dataset_name = ' / '.join(c for c in [parent_component,
                                              child_component] if c)
```